### PR TITLE
Small improvements 

### PR DESCRIPTION
**Pull request by nlybe**
- Countries input is now compatible with profile_manager plugin
- Added option to set default country in settings
- Checked compatibility with Elgg 2.x

### CODE REVIEW

File: manifest.xml [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
     <requires>
         <type>elgg_release</type>
-        <version>2.2.3</version>
+        <version>2.2</version>
```
Comment (hypeJunction) [inline manifest.xml]: Leave this as is. The only change that requires 2.2 is elgg_view_input which is unnecessary

File: languages/en.php [hunk 2]
```diff
@@ -2,404 +2,411 @@
 
 return [
 
-	'country:AD' => 'Andorra',
```
Comment (hypeJunction) [inline languages/en.php:5]: Your indentation settings are not in line with Elgg coding standards. We use 1 tab

File: manifest.xml [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
-	<id>countries</id>
+    <name>Countries</name>
+    <id>countries</id>
 
-	<description>Countries for Elgg</description>
-	<category>ui</category>
-	<version type="dist">1.1.0</version>
+    <description>Countries for Elgg</description>
+    <category>ui</category>
+    <version type="dist">2.2.0</version>
```
Comment (hypeJunction) [inline manifest.xml]: My grunt script automatically iterates the version during releases. Leave as is

File: CHANGES.md [hunk 4]
```diff
@@ -0,0 +1,13 @@
+Elgg Countries Plugin - CHANGE LOG
```
Comment (hypeJunction) [inline CHANGES.md]: My grunt script automatically creates a changelogs from commit messages. Use angular style commit messages for each feature and they will appear in the changelog

File: start.php [hunk 5]
```diff
@@ -38,4 +55,19 @@ function elgg_get_countries() {
...
 function elgg_get_country_info($fields = null, $sort_field = 'name') {
 	return \hypeJunction\Countries::getCountries('iso', $fields, $sort_field);
+}
+
+/**
+ * Get default country from settings
+ * 
+ * @return string
+ */
+function elgg_get_default_country() {
```
Comment (hypeJunction) [inline start.php]: Not needed. Just read the plugin setting in the input view

File: views/default/input/country.php [hunk 6]
```diff
@@ -7,13 +7,19 @@
...
 }
 
-echo elgg_view('input/dropdown', $vars);
+if (!isset($vars['value']) || !$vars['value']) {
+    $vars['value'] = elgg_get_default_country();
+}
+
+//echo elgg_view('input/dropdown', $vars);
+
+echo elgg_view_input('dropdown', $vars);
```
Comment (hypeJunction) [inline views/default/input/country.php]: elgg_view_input generates wrapper markup. You don't want that

File: views/default/plugins/countries/settings.php [hunk 7]
```diff
@@ -0,0 +1,21 @@
...
+ * Elgg Countries
+ * @package countries 
+ */
+	
+$plugin = $vars["entity"];
+
+$countries = elgg_get_countries();
+array_unshift($countries , elgg_echo('countries:settings:select_country'));
+
+$basic_settings = elgg_format_element('div', [], elgg_view_input('dropdown', array(
```
Comment (hypeJunction) [inline views/default/plugins/countries/settings.php]: No need to wrap. Use input/country view

**Review by hypeJunction — CHANGES_REQUESTED** [review]

**Comment by nlybe** [comment]
Followed your suggestions, hopefully

**Pull request closed without merging**